src/Tribe/Views/V2/Views/List_View.php

```diff
@@ -318,4 +318,41 @@ protected function on_page_reset() {
 		parent::on_page_reset();
 		$this->remove_past_query_args();
 	}
+
+	/**
+	 * Do a short query (one event) to determine if we should add a noindex meta tag to the page.
+	 *
+	 * @since TBD
+	 *
+	 * @param Tribe__Repository|false $events     The events repository. False by default.
+	 * @param DateTime                $start_date The start date (object) of the query.
+	 * @param Tribe__Context          $context    The current context.
+	 *
+	 * @return Tribe__Repository|false $events     The events repository results.
+	 */
+	public function get_noindex_events( $events, $start_date, $end_date, $context ) {
+		if ( null === $events ) {
```

Comment:
Here and elsewhere: refactor for early return.

```suggestion
		if ( null !== $events ) { return; }
		
		// ... do something.
```